ddtrace/internal/datadog/profiling/dd_wrapper/include/libdatadog_helpers.hpp

```diff
@@ -0,0 +1,135 @@
+    X(runtime, "runtime")                                                                                              \
+    X(runtime_id, "runtime-id")                                                                                        \
+    X(profiler_version, "profiler_version")                                                                            \
+    X(profile_seq, "profile_seq")
+
+// Here there are two columns because the Datadog backend expects these labels
+// to have spaces in the names.
+#define EXPORTER_LABELS(X)                                                                                             \
+    X(exception_type, "exception type")                                                                                \
+    X(thread_id, "thread id")                                                                                          \
+    X(thread_native_id, "thread native id")                                                                            \
+    X(thread_name, "thread name")                                                                                      \
+    X(task_id, "task id")                                                                                              \
+    X(task_name, "task name")                                                                                          \
+    X(span_id, "span id")                                                                                              \
+    X(local_root_span_id, "local root span id")                                                                        \
+    X(trace_type, "trace type")                                                                                        \
+    X(trace_resource_container, "trace resource container")                                                            \
+    X(trace_endpoint, "trace endpoint")                                                                                \
+    X(class_name, "class name")                                                                                        \
+    X(lock_name, "lock name")
+
+#define X_ENUM(a, b) a,
+#define X_STR(a, b) b,
+
+enum class ExportTagKey
+{
+    EXPORTER_TAGS(X_ENUM) Length_
+};
+
+enum class ExportLabelKey
+{
+    EXPORTER_LABELS(X_ENUM) Length_
+};
+
+inline ddog_CharSlice
+to_slice(std::string_view str)
+{
+    return { .ptr = str.data(), .len = str.size() };
+}
+
+inline std::string
+err_to_msg(const ddog_Error* err, std::string_view msg)
+{
+    auto ddog_err = ddog_Error_message(err);
+    std::string err_msg;
+    return std::string{ msg } + " (" + err_msg.assign(ddog_err.ptr, ddog_err.ptr + ddog_err.len) + ")";
+}
+
+inline std::string_view
+to_string(ExportTagKey key)
+{
+    constexpr auto num_keys = static_cast<size_t>(ExportTagKey::Length_);
+    constexpr std::array<std::string_view, num_keys> keys = { EXPORTER_TAGS(X_STR) };
```

Comment:
Minor, the invalid key could be part of the tags to help with debug when retrieving the profile.

Reply:
Indeed!  When I add logging, all of these places will emit a more nuanced/local error.

That said, this specific function _should_ be properly guarded with correct unit tests.  All of the valid tags are passed from literal expressions in the code.  It never gets generated.